resharper/resharper-unity/src/CSharp/Daemon/Stages/Analysis/StringBasedMethodInvocationProblemAnalyzer.cs

```diff
@@ -41,7 +42,11 @@ protected override void Analyze(IInvocationExpression expression, ElementProblem
             if (info.ResolveErrorType != ResolveErrorType.OK)
                 return;
 
-            var method = (info.DeclaredElement as IMethod).NotNull("info.DeclaredElement as IMethod != null");
+
+            var method = info.DeclaredElement as IMethod; // e.g. localfunction declared element, (when removing `}` and method below is parsed like local function)
```

Comment:
The comment looks strange to me. 
`IInvocationExpression` can be just a local function invocation in a correctly parsed code.
The reference expression can also resolve to almost anything e.g. fields, properties, local variables, parameters etc if they are delegates.
E.g. 
```
Action a = () => { };
a();
```

Why mention parsing recover strategy here?